lib/transit_data/glides_report/settings/load.ex

```diff
@@ -0,0 +1,60 @@
+defmodule TransitData.GlidesReport.Settings.Load do
+  @moduledoc """
+  User-selected settings for the data load step.
+  """
+
+  @type t :: %__MODULE__{
+          env_suffix: String.t(),
+          start_dt: DateTime.t(),
+          end_dt: DateTime.t(),
+          sample_rate: integer,
+          sample_count: integer | :all
+        }
+
+  defstruct [
+    :env_suffix,
+    :start_dt,
+    :end_dt,
+    :sample_rate,
+    :sample_count
+  ]
+
+  @spec new(String.t(), Date.t(), integer, integer | nil) :: t()
+  def new(env, date, sample_rate, samples_per_minute) do
+    {start_dt, end_dt} = date_to_naive_start_end(date)
+    new(env, start_dt, end_dt, sample_rate, samples_per_minute)
+  end
+
+  @spec new(String.t(), NaiveDateTime.t(), NaiveDateTime.t(), integer, integer | nil) :: t()
+  def new(env, start_dt, end_dt, sample_rate, samples_per_minute) do
+    %__MODULE__{
+      env_suffix: env,
+      start_dt: DateTime.from_naive!(start_dt, "Etc/UTC"),
+      end_dt: DateTime.from_naive!(end_dt, "Etc/UTC"),
+      sample_rate: sample_rate |> trunc(),
+      sample_count:
+        case samples_per_minute do
+          nil -> :all
+          n -> trunc(n)
+        end
+    }
+  end
+
+  defp date_to_naive_start_end(date) do
+    # We assume they want a full service day on the given date.
+    start_dt =
+      date
+      |> DateTime.new!(~T[04:00:00], "America/New_York")
+      |> DateTime.shift_zone!("Etc/UTC")
+      |> DateTime.to_naive()
```

Comment:
It seems like this is dropping the TZ information only to add it back in `load_data`. Unless this is being used elsewhere as a naive, would it make more sense to leave it in UTC?

Reply:
Sorry yeah, this is confusing. It's because `Kino.Input.utc_datetime`, when read, outputs a `NaiveDateTime` that you're supposed to assume is in UTC. I'm manually doing the same thing here so that `Settings.Load.new/5` consistently receives `NaiveDateTime`s.

Separately: `date_to_naive_start_end/1` (as well as `new/4`) is not actually used currently. I was thinking it would be nice to allow toggling between a "simple" and an "advanced" date picker, but couldn't find a straightforward, user-friendly way to implement that.

I'll do some more work in this area to at least make the code more understandable, and maybe implement that simple <-> advanced date picker toggle.